fv3core/stencils/fv_dynamics.py

```diff
@@ -160,33 +161,24 @@ def wrapup(
     )
 
 
-def fvdyn_temporaries(shape, grid):
-    origin = grid.full_origin()
+def fvdyn_temporaries(quantity_factory: fv3gfs.util.QuantityFactory, shape, grid):
     tmps = {}
-    halo_vars = ["cappa"]
-    storage_vars = ["te_2d", "dp1", "cvm"]
-    column_vars = ["gz"]
-    plane_vars = ["te_2d", "te0_2d", "wsd"]
-    utils.storage_dict(
-        tmps,
-        halo_vars + storage_vars,
-        shape,
-        origin,
-    )
-    utils.storage_dict(
-        tmps,
-        plane_vars,
-        shape[0:2],
-        origin[0:2],
-    )
-    utils.storage_dict(
-        tmps,
-        column_vars,
-        (shape[2],),
-        (origin[2],),
-    )
-    for q in halo_vars:
-        grid.quantity_dict_update(tmps, q)
+    for name in ["te_2d", "te0_2d", "wsd"]:
+        quantity = quantity_factory.empty(
+            dims=[fv3gfs.util.X_DIM, fv3gfs.util.Y_DIM], units="unknown"
+        )
+        tmps[f"{name}_quantity"] = quantity
+        tmps[name] = quantity.storage
+    for name in ["cappa", "dp1", "cvm"]:
```

Comment:
This change in convention from starting with storages and wrapping them with quantities to generating quantities and extracting the storage instead is not mentioned in the review description or explained, and seems to not have anything to do with the PR purpose. If you prefer to start using a quantity factory in this class, that's fine, but be sure to mention it. 

You also only actually need a quantity here for cappa, the rest we only use the storage for (and could potentially get removed in future refactors with stencil merging), so it's a lot of code to read for variables that are unused.

Reply:
I'll update the PR description. This was done to remove the dependency on the older `grid` class, which was previously what we used to make storages and Quantity.

In a later PR, I'm hoping to refactor these local variables and the input state into a data class with `Quantity` attributes, similarly to what @elynnwu is doing in the physics driver. The benefit of having the variables we don't halo update wrapped in Quantity is that we can access the `.dims` (and hopefully at some point `.units`) attribute to get information about it.

Reply:
Description is updated.